### ISSUE TITLE
Support views

### ISSUE THREAD

**Issue by PhilippeBerard** [joyofhex/cookbook-bind#36]
I can't use this cookbook for the moment as I need to configure **bind** with internal and external **views**.

Any plan on implementing this feature ? Thanks a lot.

**Comment by joyofhex** [comment]
We currently don't have such support but I don't think it would be complex to add. I propose to add a resource named bind_view that would configure options and match lists, and then add a 'view' parameter to various zone resources.

This would end up looking something like:

```
bind_view 'internal' do
   match_clients ['10.0.0.0/8', '192.168.0.0/16']
   options [
    'recursion yes'
   ]
end

bind_view 'external' do
   match_clients ['any']
   options [
    'recursion no'
   ]
end

bind_primary_zone 'example.com_internal' do
  name 'example.com'
  view 'internal'
end

bind_primary_zone 'example.com_external' do
  name 'example.com'
  view 'external'
end
```

Does that seem like a reasonable way of specifying the views and their configuration? If so then i'll prepare a PR this week to add that functionality.

**Comment by PhilippeBerard** [comment]
Hello David,

At first, thanks for your answer, and yes, that would be a nice way for specifying the views.

Have a nice day and happy new year !

Regards,

**Comment by joyofhex** [comment]
Happy new year to you too.

I've pushed a branch with a basic implementation of the views support. Would you like to give it a try and see if it works for you?

https://github.com/joyofhex/cookbook-bind/tree/views_support

Thanks

**Comment by joyofhex** [comment]
Released in v2.2.0.
